.github/workflows/upstream-update.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Update with upstream repository
+on:
+    schedule:
+      - cron:  0 0 * * *
+      # scheduled at 00:00 every day
+    workflow_dispatch:
+
+jobs:
+    sync_with_upstream:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: Wandalen/wretry.action@v1.0.36
+              name: Checkout
+              with:
+                  action: actions/checkout@v3
+                  with: |
+                      token: ${{ secrets.WORKFLOW_TOKEN }}
+                      ref: master
+                      fetch-depth: 100
+                  attempt_limit: 3
+                  attempt_delay: 2000
+
+            - name: Update 
+              uses: imba-tjd/rebase-upstream-action@0.5
+              with:
+                upstream: https://github.com/project-chip/connectedhomeip
+                branch: master # the upstream branch that is rebased on
+                depth: 100 # number of commits to fetch
```

Comment:
Should be OK, hopefully the upstream doesn't get more than 100 commits a day!

Reply:
I think it should be good enough. If not the job fails and we can fix it manually.

Reply:
As I understand the default is 100. IF we want to be safe, we can set it to `0`: https://github.com/imba-tjd/rebase-upstream-action/blob/master/action.yml#L40-L46 .